scwx-qt/source/scwx/qt/util/geographic_lib.cpp

```diff
@@ -28,6 +30,40 @@ const ::GeographicLib::Geodesic& DefaultGeodesic()
    return geodesic_;
 }
 
+bool GnomonicAreaContainsCenter(geos::geom::CoordinateSequence sequence)
+{
+
+   // Cannot have an area with just two points
+   if (sequence.size() <= 2 ||
+       (sequence.size() == 3 && sequence.front() == sequence.back()))
+   {
+      return false;
+   }
+   bool areaContainsPoint = false;
+   geos::geom::CoordinateXY zero {};
+   // If the sequence is not a ring, add the first point again for closure
+   if (!sequence.isRing())
+   {
+      sequence.add(sequence.front(), false);
+   }
+
+   // The sequence should be a ring at this point, but make sure
+   if (sequence.isRing())
+   {
+      try
+      {
+         areaContainsPoint =
+            geos::algorithm::PointLocation::isInRing(zero, &sequence);
+      }
+      catch (const std::exception&)
+      {
+         logger_->trace("Invalid area sequence");
```

Comment:
That code was copied from `AreaContainsPoint`.  I will update it to use a warning and `ex.what` because I do not think it is critical, but I am not sure when an error could occur there.

Note: `AreaContainsPoint` could be updated to use `GnomonicAreaContainsCenter` to eliminate code duplication.